back/engines/commercial/bulk_import_ideas/app/services/bulk_import_ideas/extractors/base_phase_extractor.rb

```diff
@@ -155,35 +159,48 @@ def text_field_attributes(column_name)
       { input_type: input_type }
     end
 
-    # NOTE: Will not work if select options contain commas or semicolons
-    # TODO: Rewrite this - override stuff is messy & only applicable to select fields right now
-    def select_field_attributes(column_name)
+    def select_field_attributes(column_name, override_input_type)
+      # Use of AI is triggered by special field prefix 'ai_'
+      use_ai = override_input_type&.start_with? 'ai_'
+      override_input_type = override_input_type.sub('ai_', '') if use_ai
+
       values = remove_empty_array_values(@idea_rows.pluck(column_name))
-      override_input_type = @config[:override_field_types][column_name]
-      maybe_multiselect = values.uniq.any? { |value| value.include?(',') || value.include?(';') } unless override_input_type == 'select'
-      values = values.map { |value| value.split(/[,;]/).map(&:strip) }.flatten if maybe_multiselect
+      unique_values = values.uniq
 
-      unique_ratio = values.size / values.uniq.size.to_f
-      if unique_ratio > 10 || %w[select multiselect].include?(override_input_type)
-        input_type = override_input_type || (maybe_multiselect ? 'multiselect' : 'select')
+      # Return a single select if specified in the config
+      return format_select_field('select', unique_values) if override_input_type == 'select'
 
-        # Update multiselect values to ensure it matches our expected format with semicolons
-        reformat_multiselect_values(column_name) if input_type == 'multiselect'
+      # Assume not a select if all values are unique
+      return nil if values.size == unique_values.size
 
-        return {
-          input_type: input_type,
-          options: values.uniq.map { |value| { title_multiloc: multiloc(value), key: generate_key(value) } }
-        }
+      # Attempt to split out multiselect values
+      multiselect_values = unique_values.map { |value| value.split(multiselect_regex).map(&:strip) }.flatten.uniq
+
+      input_type = 'select'
+      if multiselect_values.size != unique_values.size || override_input_type == 'multiselect'
+        input_type = 'multiselect'
+        unique_values = use_ai ? ai_multiselect_values(unique_values) : multiselect_values
       end
 
-      nil
+      unique_ratio = values.size / unique_values.size.to_f
```

Comment:
Makes sense, but I wonder if this works well if there are only 10 responses/inputs or so?